lib/util/aws/cloudfront_signed_url.ex

```diff
@@ -31,13 +30,43 @@ defmodule Antikythera.Aws.CloudfrontSignedUrl do
     encoded_url = if url_encoded?, do: resource_url, else: URI.encode(resource_url)
     expires_in_seconds = System.system_time(:second) + lifetime_in_seconds
     joiner = if URI.parse(encoded_url) |> Map.get(:query) |> is_nil(), do: "?", else: "&"
-    encoded_url <> joiner <> URI.encode_query(make_query_params(encoded_url, expires_in_seconds, key_pair_id, private_key))
+    encoded_url <> joiner <> URI.encode_query(make_query_params_for_canned_policy(encoded_url, expires_in_seconds, key_pair_id, private_key))
   end
 
-  defunpt make_query_params(encoded_url        :: v[String.t],
-                            expires_in_seconds :: v[pos_integer],
-                            key_pair_id        :: v[String.t],
-                            private_key        :: v[String.t]) :: [{String.t, String.t}] do
+  @doc """
+  Generates a signed URL to access a file via CloudFront using a custom policy.
+
+  ## Parameters
+
+  - `resource_url` (string): CloudFront URL used for accessing the file, including query string parameters, if any.
+  - `lifetime_in_seconds` (positive integer): Expiration time is determined as the sum of the current time (in seconds) and this value.
+  - `key_pair_id` (string): ID for an active CloudFront key pair used for generating the signature.
+  - `private_key` (string): RSA private key for the key pair specified by `key_pair_id`.
+  - `url_encoded?` (boolean): Whether `resource_url` is encoded or not (optional, default is `false`).
+  - `optional_policy` (Keyword): Optional policy conditions to be added to a custom policy (default is `[]`). Currently, supports only the following keywords:
+    - `:date_greater_than`(integer >= 0): Seconds from `AWS:EpockTime`. Specified to `DateGreaterThan`
```

Comment:
Does `AWS:EpockTime` have the same meaning to the Unix epoch?
If not, I suggest:
```suggestion
    - `:date_greater_than` (integer >= 0): Seconds from the Unix epoch. Specified to `DateGreaterThan`.
```

Reply:
CloudFront document [says](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-creating-signed-url-custom-policy.html#private-content-custom-policy-statement)
> date and time in Unix time format and UTC

Then, this should be the Unix epoch.

However, I'm not sure what will happen after [Y2038](https://en.wikipedia.org/wiki/Year_2038_problem).

Since this module is a thin wrapper for CloudFront, a user should know about CloudFront. It would be better to delegate the correct definition to AWS' document.

Reply:
I'll fix this typo.

```suggestion
    - `:date_greater_than`(integer >= 0): Seconds from `AWS:EpochTime`. Specified to `DateGreaterThan`
```

Reply:
Thank you for your explanation!

There might be misunderstanding what I meant by "the Unix epoch".
I used "the Unix epoch" as a specific time point (1970-01-01 00:00:00 UTC), not a time specification.

Since `AWS:EpockTime` is a time specification, I think changing "Seconds from `AWS:EpochTime`" to "Seconds in `AWS:EpochTime`" can avoid such confusion.

Reply:
Fixed at https://github.com/access-company/antikythera/pull/131/commits/d25bcc10ce1ab908e224886d3f4d55df36ae7744